components/script_plugins/webidl_must_inherit.rs

```diff
@@ -94,10 +92,22 @@ fn is_webidl_ty(cx: &LateContext, ty: &ty::TyS) -> bool {
 }
 
 fn check_inherits(code: &str, name: &str, parent_name: &str) -> Result<(), Box<Error>> {
-    let idl = parse_string(code)?;
-    let mut visitor = InterfaceVisitor::new(name.to_string());
-    visitor.visit(&idl);
-    let inherits = visitor.get_inherits();
+    let idl = weedle::parse(code).expect("Invalid webidl provided");
+    let mut inherits = "";
+
+    for def in idl {
+        if let weedle::Definition::Interface(def) = def {
+            if let Some(parent) = def.inheritance {
+                inherits = parent.identifier.0;
+                break;
+            }
+        } else if let weedle::Definition::CallbackInterface(def) = def {
```

Comment:
How about using pattern matching here 👀?

```rust
for def in idl {
  match def {
    weedle::Definition::Interface(def) | weedle::Definition::CallbackInterface(def) => {
      if let Some(parent) = def.inheritance {
        inherits = parent.identifier.0;
        break;
      }
    },
    _ => continue
  }
}
```

Or, maybe we can use `filter_map` with `take` here? 🤔

```rust
let inherits =
  idl.iter().filter_map(|def| {
    match def {
      weedle::Definition::Interface(def) | weedle::Definition::CallbackInterface(def) => def.inheritance,
      _ => None
    }
  })
  .take(1)
  .map_or("", |parent| parent.identifier.0);
```

Reply:
I don't think I can because the types of both `dep` is different

Reply:
Ah, I see; then please ignore me, thanks 🙇